src/fluree/db/api/branch.cljc

```diff
@@ -0,0 +1,201 @@
+
+(defn branch-info
+  "Returns detailed information about a specific branch.
+  
+  Parameters:
+    conn - Connection object
+    branch-spec - Full branch spec (e.g., \"ledger:branch\")
+    
+  Returns branch metadata including creation info, head commit, etc."
+  [conn branch-spec]
+  (go-try
+    ;; Load the ledger for this branch (which already knows its branch)
+    (let [branch-ledger (<? (connection/load-ledger conn branch-spec))]
+      ;; Get the branch info directly from the loaded ledger
+      (<? (ledger/branch-info branch-ledger)))))
+
+(defn delete-branch!
+  "Deletes a branch.
+  
+  Parameters:
+    conn - Connection object
+    branch-spec - Full branch spec to delete (e.g., \"ledger:branch\")
+    
+  Cannot delete the default branch or protected branches.
+  Returns when deletion is complete."
+  [conn branch-spec]
+  (go-try
+    (let [[_ledger-id branch] (util.ledger/ledger-parts branch-spec)
+          ;; First check if it's the main branch (handle nil as main too)
+          _ (when (or (= branch "main") (nil? branch))
+              (throw (ex-info "Cannot delete the main branch"
+                              {:status 400 :error :db/cannot-delete-main-branch})))
+          ;; Load the branch to check if it exists and is protected
+          ledger (<? (connection/load-ledger conn branch-spec))
+          ;; Get branch info to check protection status
+          branch-info (<? (ledger/branch-info ledger))
+          _ (when (:protected branch-info)
+              (throw (ex-info (str "Cannot delete protected branch: " branch)
+                              {:status 400 :error :db/cannot-delete-protected-branch})))
+          ;; Now delete the branch from nameservice
+          primary-publisher (:primary-publisher conn)]
+      (if primary-publisher
+        (do
+          (<? (nameservice/retract primary-publisher branch-spec))
+          ;; Remove from connection cache and subscriptions
+          (ns-subscribe/release-ledger conn branch-spec))
+        (throw (ex-info "No nameservice available for branch deletion"
+                        {:status 400 :error :db/no-nameservice})))
+      {:deleted branch-spec})))
+
+(defn rename-branch!
+  "Renames a branch.
+  
+  Parameters:
+    conn - Connection object
+    old-branch-spec - Current branch spec (e.g., \"ledger:old-branch\")
+    new-branch-spec - New branch spec (e.g., \"ledger:new-branch\")
+    
+  Returns when rename is complete."
+  [conn old-branch-spec new-branch-spec]
+  (go-try
+    (let [[old-ledger-id old-branch] (util.ledger/ledger-parts old-branch-spec)
+          [new-ledger-id new-branch] (util.ledger/ledger-parts new-branch-spec)]
+
+      (when (not= old-ledger-id new-ledger-id)
+        (throw (ex-info "Cannot rename branch across different ledgers"
+                        {:status 400 :error :db/invalid-branch-operation})))
+
+      (when (or (= old-branch "main") (nil? old-branch))
```

Comment:
I think this check deserves to be it's own function called `main-branch?` or similar.

Reply:
Addressed in 146c07f2b68e93971d8ed41eb00db88cefaf7cdd